src/subproblem_callback.jl

```diff
@@ -0,0 +1,57 @@
+"""
+Struct for user supbroblem callbacks.
+
+**Fields**
+
+* `name::String` : User's name for the callback. Defaults to `string(h)`.
+* `h::Function` : Callback function. See notes below for calling signature.
+* `ext::Dict{Symbol,Any}` : Dictionary to store data between callback calls or needed parameters.
+
+The callback function `h` must have the signature
+    `h(ext::Dict{Symbol,Any}, sp::SubproblemRecord, niter::Int, scenario_id::ScenarioID)::Bool`
+where `ext` is the same dictionary given to the `Callback` constructor, `sp` is a subproblem record containing a concrete type of `AbstractSubproblem` (see `SubproblemRecord` and `AbstractSubproblem`), `niter` is the current iteration and `scenario_id` is a scenario identifier (see `ScenarioID`). The callback may return `false` to stop PH.
+"""
+struct SubproblemCallback
+    name::String
+    h::Function
+    ext::Dict{Symbol,Any}
+end
+
+function Base.show(io::IO, spcb::SubproblemCallback)
+    print(io, spcb.name)
+    return
+end
+
+"""
+    SubproblemCallback(f::Function)
+
+Creates a `SubproblemCallback` structure for function `f`.
+"""
+function SubproblemCallback(f::Function)
+    return SubproblemCallback(string(f), f, Dict{Symbol,Any}())
+end
+
+"""
+    spcb(f::Function)
+
+Shorthand for `SubproblemCallback(f)`.
```

Comment:
Is it worth adding the shorthand functions for all the forms? Also a question for normal callbacks.

Reply:
In my opinion no, though I suppose `SubproblemCallback` is a bit laborious to type. I think it doesn't hurt.